[#46936] Add documentation for Rails::HealthController (#46972)
rails/rails: guides/source/action_controller_overview.md

* How to stream data directly to the user's browser.
* How to filter sensitive parameters, so they do not appear in the application's log.
* How to deal with exceptions that may be raised during request processing.

> * About the built-in health check end-point useful for load balancers and uptime monitors.

Reviewer: ```suggestion
* How to use the built-in health check end-point with load balancers and uptime monitors.
```

Should we keep all of these `How to ...`?

Author: I intentionally wanted to break up to how-to-chain :joy: basically if we are going to repeat it, then we can prefix that whole list with something that makes sense.. like:

In this guide you will learn how to:

* Stream data directly to the user's browser.
* ...

[`config.force_ssl`]: configuring.html#config-force-ssl
[`ActionDispatch::SSL`]: https://api.rubyonrails.org/classes/ActionDispatch/SSL.html

Built-in Health Check Endpoint
------------------------------

> Rails also comes with a built-in health check endpoint that is reachable at the `/up` path. This endpoint will return a 200 if the app has booted with no exceptions, otherwise a 500 status code will be returned.

Reviewer: ```suggestion
Rails also comes with a built-in health check endpoint that is reachable at the `/up` path. This endpoint will return a 200 status code if the app has booted with no exceptions, and a 500 status code otherwise.
```

> In production, many services are required to report their status upstream, whether it's to an uptime monitor that will page an engineer when things go wrong, or a load balancer or Kubernetes controller used to determine a pods health. This health check is designed to be a one-size fits all that will work in many situations.

Reviewer: ```suggestion
In production, many services are required to report their status upstream, whether it's to an uptime monitor that will page an engineer when things go wrong, or a load balancer or Kubernetes controller used to determine a pod's health. This health check is designed to be a one-size fits all that will work in many situations.
```

> For any newly generated Rails applications it will be enabled by default, but you can configure it anywhere you'd like in your "config/routes.rb":

Reviewer: ```suggestion
While any newly generated Rails applications will have the health check at `/up`, you can configure the path to be anything you'd like in your "config/routes.rb":
```
